src/buildSyncValidation.js

```diff
@@ -3,8 +3,30 @@ import Ajv from 'ajv'
 
 const setError = (errors, error) => {
     const dataPathParts = error.dataPath.split('.').slice(1)
+    //  regular expression to  get the object path and index from the ajv error data path
+    const re = /(^\w+)\[([0-9]+)\]/gm
 
     dataPathParts.reduce((errors, part, index) => {
+        let res = re.exec(part)
+        
+        if(res && res.length>0) {
+            let p = res[1]
+            let i = res[2]
+            if(!errors[p]) {
+                errors[p] = []
+            }
+            if(typeof errors[p]=='string') {
+                let err = errors[p]
+                errors[p] = []
+                errors[p]._error = err
+            }
+            if (index === dataPathParts.length -1) {
+                errors[p][i] = error.message
+            }else{
+                errors[p][i] = errors[p][i] || {}
+            }
+            return errors[p][i]
+        }
```

Comment:
Could we have a test or a description of what this part is doing? it is not obvious by reading it.

Reply:
Note that I have refactored this code. Now using only json pointers and using external libraries to do the heavy work. If there are use cases where this adaptation of the errors is failing, they should be fixed providing a test here https://github.com/Limenius/liform-react/blob/master/src/__tests__/buildSyncValidation.spec.js